deploy/helm-chart/templates/_helpers.tpl

```diff
@@ -0,0 +1,80 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "nginx-gateway.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "nginx-gateway.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Create a default fully qualified service name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "nginx-gateway.gateway.service.name" -}}
+{{- default (include "nginx-gateway.fullname" .) .Values.serviceNameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "nginx-gateway.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Common labels
+*/}}
+{{- define "nginx-gateway.labels" -}}
+helm.sh/chart: {{ include "nginx-gateway.chart" . }}
+{{ include "nginx-gateway.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "nginx-gateway.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "nginx-gateway.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+Create the name of the ServiceAccount to use
+*/}}
+{{- define "nginx-gateway.serviceAccountName" -}}
+{{- default (include "nginx-gateway.fullname" .) .Values.gateway.serviceAccount.name }}
+{{- end }}
+
+{{/*
+Expand default NGINX conf ConfigMap name.
+*/}}
+{{- define "nginx-gateway.nginx-conf" -}}
```

Comment:
I wonder if those config maps should be per release? (so they don't have static names)

I'm imagining future upgrade cases - bringing down old pod and start new ones. What if the config map update gets propagated the old pod before it shutdown (let's say, the new pod doesn't start successfully for some reason so the old one keeps running). This might break the old pod, because the new configmap might be incompatible with the old one.

At the same time, this concern might be too early to implement.

Reply:
That is a good point. However, hopefully we'll be doing away with the requirement for the config maps altogether pretty soon, after https://github.com/nginxinc/nginx-kubernetes-gateway/issues/798